doc/guides/porting/board_porting.rst

```diff
@@ -45,135 +31,333 @@ available in the source tree:
-defined by a feature set that serves the purpose of distinguishing different
-SoCs belonging to the same family.
+Once you've found an existing board that uses your SoC, you can usually start
+by copy/pasting its board directory and changing its contents for your
+hardware.
 
-SoC
----
+You need to give your board a unique name. Run ``west boards`` for a list of
+names that are already taken, and pick something new. Let's say your board is
+called ``plank`` (please don't actually use that name).
 
-Finally, an SoC is actual hardware component that is physically available on a
-board.
+Start by creating the board directory ``zephyr/boards/<ARCH>/plank``, where
+``<ARCH>`` is your SoC's architecture subdirectory. (You don't have to put your
+board directory in the zephyr repository, but it's the easiest way to get
+started. See :ref:`custom_board_definition` for documentation on moving your
+board directory to a separate repository once it's working.)
 
-Board
-=====
+Your board directory should look like this:
 
-A board implements an SoC with all its features, together with peripherals
-available on the board that differentiates the board with additional interfaces
-and features not available in the SoC.
+.. code-block:: none
 
-A board port on Zephyr typically consists of two parts:
+   boards/<ARCH>/plank
+   ├── board.cmake
+   ├── doc
+   │   └── index.rst
+   ├── Kconfig.board
+   ├── Kconfig.defconfig
+   ├── plank_defconfig
+   ├── plank.dts
+   └── plank.yaml
 
-- A hardware description (usually done by device tree), which specifies embedded
-  SoC reference, connectors and any other hardware components such as LEDs,
-  buttons, sensors or communication peripherals (USB, BLE controller, ...).
+Replace ``plank`` with your board's name, of course.
 
-- A software configuration (done using Kconfig) of features and peripheral
-  drivers. This default board configuration is subordinated to features
-  activation which is application responsibility. Though, it should also enable
-  a minimal set of features common to many applications and to applicable
-  project-provided :ref:`samples-and-demos`.
+The mandatory files are:
 
+#. :file:`plank.dts`: a hardware description in :ref:`devicetree` format. This
+   declares your SoC, connectors, and any other hardware components such as
+   LEDs, buttons, sensors, or communication peripherals (USB, BLE controller,
+   etc).
+
+#. :file:`Kconfig.board`, :file:`Kconfig.defconfig`, :file:`plank_defconfig`:
+   software configuration in :ref:`kconfig` formats. This provides default
+   settings for software features and peripheral drivers.
+
+The optional files are:
+
+- :file:`board.cmake`: used for :ref:`flash-and-debug-support`. You can also
+  use this to build source code files whenever your board is used; see
+  :ref:`build_overview` for details.
+- :file:`index.rst`: documentation for your board. You only need this if you're
```

Comment:
Perhaps you could consider adding a section that touches on the specifics of adding boards to the upstream? I think it'd be useful to set expectations for how much documentation Zephyr expects, and how to properly generate it - the Contribution Guidelines section has useful information about essential tools and standards like using Gitlint during a PR (might be good to link it) but documentation standards don't seem to be covered in an obvious spot. 

It'd be useful to know if you use a tool or IDE to generate the documentation to spec, with the correct line lengths, whitespace, etc.

Reply:
I'd also add the directories in the image section for /img/image.png, since every board has at least one.

Reply:
> Perhaps you could consider adding a section that touches on the specifics of adding boards to the upstream?

Good idea. I've added one to the end.

> I think it'd be useful to set expectations for how much documentation Zephyr expects, and how to properly generate it - the Contribution Guidelines section has useful information about essential tools and standards like using Gitlint during a PR (might be good to link it) but documentation standards don't seem to be covered in an obvious spot.

A reference to doc/templates/board.tmpl is in the new section.

That file is the single point of truth for what constitutes minimal board documentation. Any requirements you think are missing from your experience getting a board upstreamed should be added to board.tmpl instead of board_porting.rst -- please LMK if you spot anything.

> It'd be useful to know if you use a tool or IDE to generate the documentation to spec,

A reference to the documentation generation page is in the new section.

> with the correct line lengths, whitespace, etc.

The SPOT for that is the linter scripts / their configuration, which change independently from this file, so I'd rather leave those types of details out of here so they don't go stale.

Thanks for your detailed review! I think I've now addressed all your comments, please take another look.

Reply:
> Thanks for your detailed review! I think I've now addressed all your comments, please take another look.

Whoops, forgot to do the pinmux updates. Now done.